packages/core/lib/console.js

```diff
@@ -58,6 +64,26 @@ class Console extends EventEmitter {
     });
   }
 
+  // we only want to record a conflict the first time we try and load a contract
+  // into the repl context because we need to overwrite them when they change
+  recordNameConflicts(abstractions) {
+    for (const abstraction of abstractions) {
+      const globals = Object.getOwnPropertyNames(this.repl.context.global);
+      const name = abstraction.contract_name;
+      if (this.replContextNameConflicts[name] === undefined) {
+        const conflict = globals.includes(name);
+        // note that we need to warn the user later
+        if (conflict) {
+          this.warnUserOfConflicts = true;
```

Comment:
This could be an actual collection that builds the new contracts, that can be used to alert, as well as update the known collisions.